src/Psr/Tracing/RelayOpenTelemetry.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CacheWerk\Relay\Psr\Tracing;
+
+use Relay\Relay;
+
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\TracerInterface;
+
+class RelayOpenTelemetry
+{
+    /**
+     * The Relay instance.
+     *
+     * @var \Relay\Relay
+     */
+    protected Relay $relay;
+
+    /**
+     * The OpenTelemetry tracer instance.
+     *
+     * @var \OpenTelemetry\API\Trace\TracerInterface
+     */
+    protected TracerInterface $tracer;
+
+    /**
+     * Creates a new `NewRelicRelay` instance.
+     *
+     * @param  \Relay\Relay  $relay
+     * @param  \OpenTelemetry\API\Trace\TracerInterface  $tracer
+     * @return void
+     */
+    public function __construct(Relay $relay, TracerInterface $tracer)
+    {
+        $this->relay = new $relay;
+        $this->tracer = new $tracer;
+    }
+
+    /**
+     * Executes Relay methods inside OpenTelemetry span.
+     *
+     * @param  string  $name
+     * @param  array  $arguments
+     * @return mixed
+     */
+    public function __call(string $name, array $arguments)
+    {
+        $span = $this->tracer->spanBuilder('Relay') // TODO: What's the name?
+            ->setAttribute('command', strtolower($name))
+            ->setSpanKind(SpanKind::KIND_CLIENT) // TODO: Is this the correct kind?
```

Comment:
Yes, this is correct. Clients like Relay use the client kind and servers like Redis Server or the Relay backend use the server kind.